src/Chainweb/Pact/PactService.hs

```diff
@@ -349,16 +364,25 @@ logDebug = logg "DEBUG"
 
 -- | Validate a mined block.  Execute the transactions in Pact again as validation
 -- | Note: The BlockHeader here is the header of the block being validated
-execValidateBlock :: Bool -> BlockHeader -> PayloadData -> PactServiceM PayloadWithOutputs
-execValidateBlock loadingGenesis currHeader plData = do
-
-    miner <- decodeStrictOrThrow (_minerData $ _payloadDataMiner plData)
-
+execValidateBlock
+    :: MemPoolAccess
+    -> Bool
+    -> BlockHeader
+    -> PayloadData
+    -> PactServiceM PayloadWithOutputs
+execValidateBlock mpAccess loadingGenesis currHeader plData = do
     let bHeight@(BlockHeight bh) = _blockHeight currHeader
-        bParent = _blockParent currHeader
         bHash = _blockHash currHeader
+        bParent = _blockParent currHeader
         isGenesisBlock = isGenesisBlockHeader currHeader
 
+    logInfo $ "execValidateBlock, about to get call setLastHeader: "
+        <> " (height = " <> sshow bHeight <> ")"
+        <> " (hash = " <> sshow bHash <> ")"
+    liftIO $ mpaSetLastHeader mpAccess currHeader
```

Comment:
You can do this inside `processFork` ?

Reply:
Not when processFork is being called only as part of the newBlock process.  This most importantly needs to happen at validate time